lib/logstash/inputs/cloudwatch_logs.rb

```diff
@@ -110,16 +110,21 @@ def check_start_position_validity
   public
   def run(queue)
     @queue = queue
+    @priority = []
     _sincedb_open
     determine_start_position(find_log_groups)
 
     while !stop?
-      groups = find_log_groups
-
-      groups.each do |group|
-        @logger.debug("calling process_group on #{group}")
-        process_group(group)
-      end # groups.each
+      begin
+        groups = find_log_groups
```

Comment:
`find_log_groups` hits the API as well which can be throttled. If it's not handled then it'll simply error and cause the whole plugin to reload (potentially losing position etc). With this change throttling is always handled for all API calls.